src/prommis/solvent_extraction/tests/test_sx_model_dynamic.py

```diff
@@ -0,0 +1,295 @@
+    m.fs.solex.mscontactor.aqueous[0, :].conc_mass_comp["Sm"].fix(0.097)
+    m.fs.solex.mscontactor.aqueous[0, :].conc_mass_comp["Gd"].fix(0.2584)
+    m.fs.solex.mscontactor.aqueous[0, :].conc_mass_comp["Dy"].fix(0.047)
+
+    m.fs.solex.mscontactor.aqueous_inherent_reaction_extent[0, :, "Ka2"].fix(0)
+    m.fs.solex.mscontactor.aqueous[0, :].flow_vol.fix(62.01)
+
+    m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Al"].fix(1e-7)
+    m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Ca"].fix(1e-7)
+    m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Fe"].fix(1e-7)
+    m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Sc"].fix(1e-7)
+    m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Y"].fix(1e-7)
+    m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["La"].fix(1e-7)
+    m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Ce"].fix(1e-7)
+    m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Pr"].fix(1e-7)
+    m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Nd"].fix(1e-7)
+    m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Sm"].fix(1e-7)
+    m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Gd"].fix(1e-7)
+    m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Dy"].fix(1e-7)
+
+    m.fs.solex.mscontactor.organic[0, :].flow_vol.fix(62.01)
+
+    return m
+
+
+@pytest.mark.unit
+def test_structural_issues(model):
+    dt = DiagnosticsToolbox(model)
+    dt.report_structural_issues()
+    dt.assert_no_structural_warnings(ignore_unit_consistency=True)
+
+
+@pytest.mark.component
+@pytest.mark.solver
+def test_solve(model):
+    solver = SolverFactory("ipopt")
+    results = solver.solve(model, tee=False)
+
+    assert_optimal_termination(results)
+
+
+@pytest.mark.component
+@pytest.mark.solver
+def test_numerical_issues(model):
+    dt = DiagnosticsToolbox(model)
+    dt.assert_no_numerical_warnings()
+
+
+@pytest.mark.component
+@pytest.mark.solver
+def test_solution(model):
+    time_duration = 24
+    aqueous_outlet = {
+        "H2O": 1000000,
+        "H": 1.75564,
+        "SO4": 3999.8761,
+        "HSO4": 693.4001,
+        "Al": 363.321,
+        "Ca": 81.9851,
+        "Cl": 9.9999e-8,
+        "Ce": 0.57801,
+        "Dy": 3.4488e-03,
+        "Fe": 461.7292,
+        "Gd": 0.07428,
+        "La": 0.40433,
+        "Nd": 9.7359e-04,
+        "Pr": 0.09789,
+        "Sc": 1.3159e-03,
+        "Sm": 9.9817e-05,
+        "Y": 1.27602e-4,
+    }
+
+    organic_outlet = {
+        "Al": 60.42,
+        "Ca": 27.923,
+        "Ce": 1.7771,
+        "Dy": 0.045844,
+        "Fe": 235.631,
+        "Gd": 0.19347,
+        "La": 0.6021,
+        "Nd": 0.9821,
+        "Pr": 0.21488,
+        "Sc": 1.76797,
+        "Sm": 0.10081,
+        "Y": 0.12888,
+    }
+
+    for k, v in model.fs.solex.organic_outlet.conc_mass_comp.items():
+        if k[0] == time_duration:
+            assert value(v) == pytest.approx(organic_outlet[k[1]], rel=1e-4)
```

Comment:
Some changes have been made since the last time Andrew commented on this, so would this test pass now if you changed the relative tolerance to 1e-5?

Reply:
The assertion test is failing at the 4th decimal place,  Obtained: 0.6020663510939577  Expected: 0.6021 ± 6.0e-06 , otherwise there is no big change in the values. Do you want me to update the values?

Reply:
I don't think any changes are necessary here then

Reply:
Okay